R/featureMethods.R

```diff
@@ -0,0 +1,246 @@
+#' ## Filter complex features by a peak correlation of 0.5, a minimum hypothesis
+#' ## completeness of 0.5 and a minimum distance to the monomers by a factor of 2:
+#' filteredComplexFeatures <- filterFeatures(complexFeatures, 
+#'                                      min_peak_corr=0.5, 
+#'                                      min_hypothesis_completeness=0.5,
+#'                                      min_monomer_distance_factor=2)
+#' ## Run summary function on filtered data:
+#' summarizeFeatures(filteredComplexFeatures)
+#' @export
+filterFeatures <- function(feature_table,
+                           complex_ids=NULL,
+                           protein_ids=NULL,
+                           min_feature_completeness=NULL,
+                           min_hypothesis_completeness=NULL,
+                           min_subunits=NULL,
+                           min_peak_corr=0.5,
+                           min_monomer_distance_factor=NULL){
+  if("complex_id" %in% names(feature_table)){
+    type="complex"
+  } else if ("protein_id" %in% names(feature_table)) {
+    type="protein"
+  } else {
+    stop("This is neither a complex or a protein feature table. Please check the input data.")
+  }
+  if(!is.null(complex_ids)){
+    if(type=="complex"){
+      feature_table <- subset(feature_table,complex_id %in% as.character(complex_ids))
+    } else {
+      stop("This is not a complex feature table. Please check the input data and your filtering criteria, especially the complex_ids.")
+    }
+  }
+  if(!is.null(protein_ids)){
+    if(type=="protein"){
+      feature_table <- subset(feature_table,protein_id %in% as.character(protein_ids))
+    } else {
+      stop("This is not a protein feature table. Please check the input data and your filtering criteria, especially the protein_ids.")
+    }
+  }
+  if(!is.null(min_feature_completeness)){
+    feature_table <- subset(feature_table,completeness >= min_feature_completeness)
+  }
+  if(!is.null(min_hypothesis_completeness)){
+    if(type=="complex"){
+      allowed_ids <- feature_table[completeness>=min_hypothesis_completeness, unique(complex_id)]
+      feature_table <- feature_table[complex_id %in% allowed_ids]
+    } else if (type=="protein") {
+      allowed_ids <- feature_table[completeness>=min_hypothesis_completeness, unique(protein_id)]
+      feature_table <- feature_table[protein_id %in% allowed_ids]
+    }
+  }
+  if(!is.null(min_subunits)){
+    feature_table <- subset(feature_table,n_subunits_detected >= min_subunits)
+  }
+  if(!is.null(min_peak_corr)){
+    feature_table <- subset(feature_table,peak_corr >= min_peak_corr)
+  }
+  if(!is.null(min_monomer_distance_factor)){
+    if (! "monomer_mw" %in% names(feature_table)){
+      stop("No molecular weight information available. The min_monomer_distance_factor is only a valid 
+           filtering option if molecular weight information of the proteins is available.")
+    }
+    dist = lapply(seq(1:nrow(feature_table)), function(i){
```

Comment:
use <-

also this would be neater with apply(feature_table, 1, FUN...) but not that important